Tests/OpenCombineTests/PublisherTests/FutureTests.swift

```diff
@@ -204,4 +204,25 @@ final class FutureTests: XCTestCase {
             .completion(.failure(error))
         ])
     }
+
+    func testStartsImmediately() {
+        var hasStarted = false
+        _ = Sut { _ in hasStarted = true }
+        XCTAssertTrue(hasStarted)
+    }
+
+    func testWaitsForRequest() {
+        var promise: Sut.Promise?
+
+        let future = Sut { promise = $0 }
+
+        let subscriber = TrackingSubscriber()
+        future.subscribe(subscriber)
+
+        promise?(.success(42))
+
+        XCTAssertEqual(subscriber.history, [
+            .subscription("Future")
+        ])
+    }
```

Comment:
Let's catch the subscription:

```swift
var downstreamSubscription: Subscription?
let subscriber = TrackingSubscriber(
    receiveSubscription: { downstreamSubscription = $0 }
)
```

and later make sure that after requesting `.max(1)` the value is sent downstream.